File: app/src/main/java/com/salah/tracker/data/database/entities/PrayerLog.kt
package com.salah.tracker.data.database.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "prayer_logs")
data class PrayerLog(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val date: String, // format: "yyyy-MM-dd"
    val prayerName: String, // Fajr, Dhuhr, Asr, Maghrib, Isha, Tahajjud, Sunnah, Nafl
    val status: String, // Offered On-Time, Offered Late, Missed/Qaza, Excused, Pending
    val loggedTimestamp: Long? = null
)

File: flutter_app/lib/data/models/prayer_log.dart
class PrayerLog {
  final int? id;
  final String date;
  final String prayerName;
  final String status;
  final int? loggedTimestamp;

  PrayerLog({
    this.id,
    required this.date,
    required this.prayerName,
    required this.status,
    this.loggedTimestamp,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'date': date,
      'prayerName': prayerName,
      'status': status,
      'loggedTimestamp': loggedTimestamp,
    };
  }

  factory PrayerLog.fromMap(Map<String, dynamic> map) {
    return PrayerLog(
      id: map['id'],
      date: map['date'],
      prayerName: map['prayerName'],
      status: map['status'],
      loggedTimestamp: map['loggedTimestamp'],
    );
  }
}
